lib/nrf_modem_lib/nrf_modem_lib.c

```diff
@@ -32,16 +32,6 @@ BUILD_ASSERT(IPC_IRQn == NRF_MODEM_IPC_IRQ, "NRF_MODEM_IPC_IRQ mismatch");
  */
 #define NRF_MODEM_LIB_SHMEM_TX_HEAP_OVERHEAD_SIZE 128
 
-struct shutdown_thread {
-	sys_snode_t node;
-	struct k_sem sem;
-};
-
-static sys_slist_t shutdown_threads;
-static bool first_time_init;
-static struct k_mutex slist_mutex;
-
-static int init_ret;
```

Comment:
nit: this can be removed in the first commit (`lib: nrf_modem: remove deprecated nrf_modem_lib_get_init_ret()`)